Assets/Scripts/EnemySpawner.cs

```diff
@@ -0,0 +1,58 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class EnemySpawner : MonoBehaviour
+{
+
+    [SerializeField] private List<Enemies> enemiesToSpawn;
+
+    private int maxEnemies = 10;
+    private float spawnRate = 3;
+    private float timer = 0;
+
+    private float maxOffset = 8;
+    private float minOffset = -8;
+
+    [SerializeField] private List<Enemies> enemiesAlive;
+
+    private Action OnEnemiesKilled;
+
+    private void Start()
+    {
+        
+    }
+
+    private void Update()
+    {
+        if (timer < spawnRate)
+        {
+            timer += Time.deltaTime;
+        }
+        else
+        {
+            if (enemiesAlive.Count < maxEnemies)
+            {
+                SpawnEnemy();
+                timer = 0;
+            }
+        }
```

Comment:
Add an if to reset the timer to 0 if reaches float.maxValue

Reply:
Done